watson_developer_cloud/visual_recognition_v3.py

```diff
@@ -12,365 +14,1331 @@
+a custom classifier to identify subjects that suit your needs.
+
+**Tip**: To test calls to the **Custom classifiers** methods with the API explorer,
+provide your `api_key` from your Bluemix service instance.
 """
+
+from __future__ import absolute_import
+
 import json
-import mimetypes
+from .watson_service import WatsonService
 
-from .watson_developer_cloud_service import WatsonDeveloperCloudService
+##############################################################################
+# Service
+##############################################################################
 
 
-class VisualRecognitionV3(WatsonDeveloperCloudService):
-    """Client for the Visual Recognition service"""
+class VisualRecognitionV3(WatsonService):
+    """The Visual Recognition V3 service."""
 
     default_url = 'https://gateway-a.watsonplatform.net/visual-recognition/api'
-    latest_version = '2016-05-20'
+    VERSION_DATE_2016_05_20 = '2016-05-20'
 
-    def __init__(self, version, url=default_url, **kwargs):
+    def __init__(self, version, url=default_url, api_key=None):
         """
-        Construct an instance. Fetches service parameters from VCAP_SERVICES
-        runtime variable for Bluemix, or it defaults to local URLs.
-        :param version: specifies the specific version-date of the service to
-        use, for example '2016-05-20'
-        :param api_key: specifies the credentials for the service (doesn't
-        use username and password)
+        Construct a new client for the Visual Recognition service.
+
+        :param str version: The API version date to use with the service, in
+               "YYYY-MM-DD" format. Whenever the API is changed in a backwards
+               incompatible way, a new minor version of the API is released.
+               The service uses the API version for the date you specify, or
+               the most recent version before that date. Note that you should
+               not programmatically specify the current date at runtime, in
+               case the API has been updated since your application's release.
+               Instead, specify a version date that is compatible with your
+               application, and don't change it until your application is
+               ready for a later version.
+
+        :param str url: The base url to use when contacting the service (e.g.
+               "https://gateway.watsonplatform.net/visual-recognition/api").
+               The base url may differ between Bluemix regions.
+
+        :param str api_key: The API Key used to authenticate.
+
         """
 
-        WatsonDeveloperCloudService.__init__(self, 'watson_vision_combined',
-                                             url, **kwargs)
+        WatsonService.__init__(
+            self,
+            vcap_services_name='watson_vision_combined',
+            url=url,
+            api_key=api_key,
+            use_vcap_services=True)
```

Comment:
Part of the `**kwargs` issue, but we should still let users set this false when needed, not just hardcode it as true